test/refactoring/ConvertAsync/basic.swift

```diff
@@ -0,0 +1,189 @@
+func errorNonOptionalResult(completion: (String, Error?) -> Void) { }
+// ASYNC-ERRORNONOPTIONALRESULT: func errorNonOptionalResult() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-CUSTOMERROR %s
+func customError(completion: (String?, CustomError?) -> Void) { }
+// ASYNC-CUSTOMERROR: func customError() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-ALIAS %s
+func alias(completion: SomeCallback) { }
+// ASYNC-ALIAS: func alias() async -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-NESTEDALIAS %s
+func nestedAlias(completion: NestedAliasCallback) { }
+// ASYNC-NESTEDALIAS: func nestedAlias() async -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-SIMPLERESULT %s
+func simpleResult(completion: (Result<String, Never>) -> Void) { }
+// ASYNC-SIMPLERESULT: func simpleResult() async -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-ERRORRESULT %s
+func errorResult(completion: (Result<String, Error>) -> Void) { }
+// ASYNC-ERRORRESULT: func errorResult() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-CUSTOMERRORRESULT %s
+func customErrorResult(completion: (Result<String, CustomError>) -> Void) { }
+// ASYNC-CUSTOMERRORRESULT: func customErrorResult() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ASYNC-ALIASRESULT %s
+func aliasedResult(completion: SomeResultCallback) { }
+// ASYNC-ALIASRESULT: func aliasedResult() async throws -> String { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=MANY %s
+func many(_ completion: (String, Int) -> Void) { }
+// MANY: func many() async -> (String, Int) { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=OPTIONAL-SINGLE %s
+func optionalSingle(completion: (String?) -> Void) { }
+// OPTIONAL-SINGLE: func optionalSingle() async -> String? { }
+
+// RUN: %refactor -add-async-alternative -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=MANY-OPTIONAL %s
+func manyOptional(_ completion: (String?, Int?) -> Void) { }
+// MANY-OPTIONAL: func manyOptional() async -> (String?, Int?) { }
+
+// 2. Check that the next functions do not have async alternatives
```

Comment:
Thanks, I actually have 2 but they got put into the next commits somehow :P. I'll add the rest

Reply:
I've added this: the generic case is currently not ideal, the generic parameter will still exist even if it's not used. It would be nice if we could run fixits on the generated code, would fix the force unwrap case as well (that has its own handling too).